### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by tonyrex99** [react-auth-kit/react-auth-kit#1353]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

**Describe the bug** \
I'm trying to use react-auth-kit in my react app but I keep getting this error

history.ts:480 Uncaught Error: useLocation() may be used only in the context of a <Router> component.
    at invariant (history.ts:480:11)
    at useLocation (hooks.tsx:102:3)
    at RequireAuth (PrivateRoute.tsx:51:22)
    at renderWithHooks (react-dom.development.js:16305:18)
    at mountIndeterminateComponent (react-dom.development.js:20074:13)
    at beginWork (react-dom.development.js:21587:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at beginWork$1 (react-dom.development.js:27451:7)
    
    

**To Reproduce** \
import React from "react";
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from "react-router-dom";
import Dashboard from "./pages/Dashboard.jsx";
import "./index.css";
import "./global.css";
import ErrorPage from "./error-page.jsx";
import Statistics from "./pages/Statistics.jsx";
import Overview from "./pages/Overview.jsx";
import { Products } from "./pages/Products.jsx";
import { Profile } from "./pages/Profile.jsx";
import { Feedbacks } from "./pages/Feedbacks.jsx";
import { Store } from "./pages/Store.jsx";
import { Orders } from "./pages/Orders.jsx";
import { Customers } from "./pages/Customers.jsx";
import store from "./app/store.js";
import { Provider } from "react-redux";
import { AuthPage } from "./pages/auth/index.tsx";
import RegisterLeft from "./RegisterLeft.jsx";
import WelcomeLeft from "./WelcomeLeft.jsx";
import otpImage from "./assets/slideshow/slide5.png";
import finalSlide from "./assets/slideshow/successSlide.png";
import successTick from "./assets/slideshow/successTick.png";
import { AuthProvider, RequireAuth } from "react-auth-kit";

const App = () => (
  <AuthProvider
    authType="cookie"
    authName="_auth"
    cookieDomain={window.location.hostname}
    cookieSecure={false}
  >
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<Navigate to="/login" replace={true} />} />
          <Route path="dashboard" element={<Dashboard />}>
            <Route path="" element={<Overview />} />
            <Route path="overview" element={<Overview />} />
            <Route
              path="statistics"
              element={
                <RequireAuth loginPath="/login">
                  <Statistics />
                </RequireAuth>
              }
            />
            <Route path="products" element={<Products />} />
            <Route path="profile" element={<Profile />} />
            <Route path="store" element={<Store />} />
            <Route path="orders" element={<Orders />} />
            <Route path="feedbacks" element={<Feedbacks />} />
            <Route path="customers" element={<Customers />} />
          </Route>
          <Route
            path="login"
            element={
              <AuthPage
                leftPane={<WelcomeLeft />}
                title={"Gazzar"}
                formProps={{
                  initialValues: {
                    email: "demo@refine.dev",
                    password: "demodemo",
                  },
                }}
              />
            }
          />
          <Route
            path="forgot-password"
            element={
              <AuthPage
                type={"forgotPassword"}
                title={"Gazzar"}
                leftPane={<WelcomeLeft />}
                formProps={{
                  initialValues: {
                    email: "demo@refine.dev",
                    password: "demodemo",
                  },
                }}
              />
            }
          />
          <Route
            path="update-password"
            element={
              <AuthPage
                type={"updatePassword"}
                title={"Gazzar"}
                leftPane={<WelcomeLeft />}
                formProps={{
                  initialValues: {
                    email: "demo@refine.dev",
                    password: "demodemo",
                  },
                }}
              />
            }
          />
          <Route
            path="otp"
            element={
              <AuthPage
                type={"otpVerification"}
                title={"Gazzar"}
                successContent={{
                  description: (
                    <div>
                      Thank you for registering with Gazzar.
                      <br />
                      Your will be notified once your account has been <br />{" "}
                      successfully verified{" "}
                    </div>
                  ),
                  buttonText: "Done",
                  image: successTick,
                }}
                successSlide={
                  <WelcomeLeft
                    key={"success"}
                    content={[
                      {
                        image: finalSlide,
                        title: (
                          <div>
                            You're all
                            <br /> ready to go!{" "}
                          </div>
                        ),
                        width: 114,
                        height: 214,
                      },
                    ]}
                  />
                }
                leftPane={
                  <WelcomeLeft
                    key={"otppage"}
                    content={[
                      {
                        image: otpImage,
                        title: (
                          <div>
                            Hang on, <br /> We're almost there{" "}
                          </div>
                        ),
                        width: "380px",
                        height: "268px",
                      },
                    ]}
                  />
                }
                email="email@website.com"
                inputCount={4}
                numberPerCell={1}
              />
            }
          />
          <Route
            path="register"
            element={
              <AuthPage
                type={"register"}
                title={"Gazzar"}
                leftPane={<RegisterLeft />}
                contentProps={{
                  style: {
                    alignSelf: "center",
                    marginTop: 50,
                  },
                }}
                formProps={{
                  initialValues: {
                    email: "demo@refine.dev",
                    password: "demodemo",
                  },
                }}
              />
            }
          >
            <Route
              path="otp"
              element={
                <AuthPage
                  type={"otpVerification"}
                  title={"Gazzar"}
                  leftPane={<WelcomeLeft />}
                  formProps={{
                    initialValues: {
                      email: "demo@refine.dev",
                      password: "demodemo",
                    },
                  }}
                />
              }
            />
          </Route>
        </Routes>
      </Router>
    </Provider>
  </AuthProvider>
);

export default App;

**Expected behavior** \
it navigates me back to login page

**Current behaviour** \
blank page with an error

**Possible solutions** (Optional) \
Give ideas for solving the problem, provide relevant links to additional packages you are mentioning

## License

[Apache-2.0](https://github.com/react-auth-kit/react-auth-kit/blob/master/LICENSE) © [React Auth Kit](https://github.com/react-auth-kit/react-auth-kit)

**Comment by poshl9k** [comment]
Similar here with RequireAuth component

ERROR
useLocation() may be used only in the context of a <Router> component.


`    <BrowserRouter>
      <RoutesComponent>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route path="objects">
              <Route
                path=""
                element={
                  <RequireAuth loginPath="auth">
                    <Objects />
                  </RequireAuth>
                }
              />
              <Route path=":guid" element={<Object needMenu={true} />} />
            </Route>
            <Route path="users/me/" element={<UserPage />} />
          </Route>
          <Route path="register" exact element={<Register />} />
          <Route path="auth" exact element={<Auth />} />
        </Routes>
      </RoutesComponent>
    </BrowserRouter>`

**Comment by haonguyen22** [comment]
I made the same mistake and noticed the RequireLogin note indicating that only React-Router v6.x should be used. As a result, I downgraded the react-auth-kit version to v1.x because I currently use React-Router v5.x, and everything worked as expected. 
I hope this information is helpful to you.
![image](https://github.com/react-auth-kit/react-auth-kit/assets/83975213/b8b71107-4b58-4779-a406-526622a9dc37)
